confd/pkg/backends/calico/client.go

```diff
@@ -1355,7 +1394,11 @@ func (c *client) updateCache(updateType api.UpdateType, kvp *model.KVPair) bool
 		c.cache[k] = newValue
 	}
 
-	log.Debugf("Cache entry updated from event type %d: %s=%s", updateType, k, c.cache[k])
+	logVal := c.cache[k]
+	if c.isSensitive(k) {
```

Comment:
sigh, yeah this is necessary but icky. I guess we didn't need this for global peers because we don't store the password in the cache in the same way so it was never logged?

Reply:
Yup, BGP peer updates didn't go through this same logic so those updates were never logged. I thought about creating a "sanitized" method for updating objects in cache, but it's effectively a copy of the code just with this extra logic in it and the code duplication seemed worse than just doing this.